core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1635,24 +1635,6 @@ class Partition(val topicPartition: TopicPartition,
     localLog.fetchOffsetSnapshot
   }
 
-  def legacyFetchOffsetsForTimestamp(timestamp: Long,
-                                     maxNumOffsets: Int,
-                                     isFromConsumer: Boolean,
-                                     fetchOnlyFromLeader: Boolean): Seq[Long] = inReadLock(leaderIsrUpdateLock) {
-    val localLog = localLogWithEpochOrThrow(Optional.empty(), fetchOnlyFromLeader)
-    val allOffsets = localLog.legacyFetchOffsetsBefore(timestamp, maxNumOffsets)
```

Comment:
Perhaps it can be removed from `LocalLog` too (in a follow-up).

Reply:
You are right.
If this PR gets merged, this method would only be invoked in tests.
I will file a JIRA for this one.